### PR TITLE
Query params on Favorites page

### PR DESCRIPTION
**Pull request by mjoyce91**
Imports the [use-query-params](https://github.com/pbeshai/use-query-params) library and uses it on the Favorites page. This enables the browser to track the history of the user's choices, resulting in their filters persisting after viewing a position and clicking "Back".

### CODE REVIEW

**Comment by mjoyce91** [comment]
@elizabeth-jimenez plz find all the edge cases 😃

File: src/reducers/favoritePositions/favoritePositions.js [hunk 1]
```diff
@@ -33,3 +33,15 @@ export function favoritePositions(state = {
       return state;
   }
 }
+
+export function favoritePositionsPreferences(state = {}, action) {
```
Comment (elizabeth-jimenez) [inline src/reducers/favoritePositions/favoritePositions.js:37]: Discussed in Discord: Doesn't appear to be being used anywhere.

File: src/actions/favoritePositions.js [hunk 2]
```diff
@@ -42,6 +42,13 @@ export function favoritePositionsFetchDataSuccess(results) {
   };
 }
 
+export function favoritePositionsPreferences(preferences) {
+  return {
+    type: 'FAVORITE_POSITIONS_PREFERENCES',
+    preferences,
+  };
+}
+
```
Comment (elizabeth-jimenez) [inline src/actions/favoritePositions.js:51]: Discussed in Discord: Doesn't appear to be being used anywhere.

File: src/Containers/Favorites/Favorites.jsx [hunk 3]
```diff
@@ -125,6 +114,8 @@ FavoritePositionsContainer.defaultProps = {
   favoritePositionsIsLoading: false,
   bidList: { results: [] },
   userProfileFavoritePositionIsLoading: new Set(),
+  query: {},
+  setQuery: EMPTY_FUNCTION,
 };
 
 FavoritePositionsContainer.contextTypes = {
```
Comment (elizabeth-jimenez) [inline src/Containers/Favorites/Favorites.jsx:121]: Discussed in Discord: i know you didn't add the contextTypes 😅  but we don't seem to use it anymore

**Comment by elizabeth-jimenez** [comment]
Discussed already in Discrod: `sortType` is getting updated in the `query` but the actual dropdown doesn't seem to be reading it

File: src/Containers/Favorites/Favorites.jsx [hunk 4]
```diff
@@ -14,70 +15,57 @@ import CompareDrawer from 'Components/CompareDrawer';
...
     if (get(e, 'page', 1) !== page) {
-      setPage(e.page);
+      setPage$(e.page);
       scrollToTop({ delay: 0, duration: 400 });
     }
   }
 
   function selectedNav(navVal) {
-    setNavType(navVal);
+    setNavType$(navVal);
```
Comment (elizabeth-jimenez) [inline src/Containers/Favorites/Favorites.jsx:63]: 💰

File: src/Containers/Favorites/Favorites.jsx [hunk 5]
```diff
@@ -14,70 +15,57 @@ import CompareDrawer from 'Components/CompareDrawer';
...
+      // Only fetch all if counts doesn't exist; otherwise fetch selected navType
+      let type = 'all';
+      if (keys(favoritePositions.counts).length) {
+        type = navType;
+      }
+      setCalled(true);
+      props.bidListFetchData();
+      getFavorites(type);
     }
-  }, [userProfileFavoritePositionIsLoading]);
```
Comment (elizabeth-jimenez) [inline src/Containers/Favorites/Favorites.jsx:61]: 😢

**Review by elizabeth-jimenez — APPROVED** [review]
Works really well! 🍰 🥮 🍬 🍭 👍

**Pull request merged**